compiler/rustc_infer/src/infer/opaque_types/mod.rs

```diff
@@ -610,6 +610,15 @@ impl<'tcx> InferCtxt<'tcx> {
             ty::ClauseKind::WellFormed(hidden_ty.into()),
         ));
 
+        // This enforces the region obligations that link the bivariant
+        // lifetimes of an opaque to their invariant copies.
+        obligations.push(traits::Obligation::new(
+            tcx,
+            cause.clone(),
+            param_env,
+            ty::ClauseKind::WellFormed(Ty::new_opaque(self.tcx, def_id, args).into()),
+        ));
```

Comment:
is this necessary? I'd expect that we require the opaque to be well-formed in the location which introduces it. We have to check the hidden type, as the structural recursion in `wf.rs` does not skip it, so there are types in "relevant positions" which are not wf.

Reply:
is #122078 already included in your local branch? I would expect this to be necessary until then.

Reply:
it also doesn't check that the args are wf for functions :thinking: going to quickly put up a PR here